### ISSUE TITLE
[low prio] ui middleware does not bind properly to some paths.

### ISSUE THREAD

**Issue by croraf** [carlansley/swagger2-koa#14]
I have the following setup:
```
app.use(cors());
app.use(koaBody());
app.use(validate(document));

router.get('api/path/', ...);
router.post('api/path/', ...);

app.use(router.routes());

app.use(ui(document, '/api'));

app.use(async (ctx) => {
    let urlPath = ctx.path;
    if (ctx.path === '/') urlPath = '/index.html';
    await send(ctx, urlPath, { root: path.resolve(__dirname, '../../frontend') });
});
```

When I try to access swagger with `host/api` I get 404 not found. 
What is the reason for that (does router bind to this path)?

When I change `app.use(ui(document, '/api'));` to` app.use(ui(document));` and try to access swagger with `host/` I get: `Can't read swagger JSON from http://localhost:3000/api-docs`

So 'host/' resource is received but when it tries to access the `api-docs` resource it fails with 404 not found. It looks like `ui` does not bind to that path.

Finally, I managed to make it running by binding `ui `to some other root path like `/swagger`.

So the two questions are why cannot I bind to '/' or '/api'?

**Comment by carlansley** [comment]
Whats the `basePath` set to in your swagger file?  The router will prepend that value to the path.

**Comment by croraf** [comment]
It is /api, but `ui ` function does not append swagger to that. If I set `app.use(ui(document, '/swagger'))` the swagger is available at localhost/swagger, nothing to do with /api.

But if I  set `app.use(ui(document))` I cannot access swagger at `localhost/` nor at `localhost/api` as explained above.

**Comment by croraf** [comment]
Perhaps` app.use(validate(document));` binds to that prefix, and catches everything on that prefix, so that's why I cannot set swagger-ui to `app.use(ui(document, '/api'));`

But what is the reason for not beeing able to bind to `app.use(ui(document))` and access it with `localhost/`

**Comment by croraf** [comment]
OK I found the reason why it doesn't work in `app.use(ui(document))` case. 

So to access swagger I need to use `localhost/` url, this immediately gets me swagger html page that you call `uiHtml` in the code. 

But then this page needs to require `/api-docs`.  As this resource starts with `/api...` validate middleware catches it and sees no valid resource that has `...-docs` and rejects it with 404.

**Comment by carlansley** [comment]
Thanks for the bug report, fixed in 0.0.37
